### PR TITLE
Migrate Integration Tests

### PR DESCRIPTION
**Pull request by jeff-hernandez**
The integration tests will be added as a GitHub Action. We'll need to add secrets in the repository settings for `PP_K` to run premium primitive tests.

### CODE REVIEW

**Comment by jeff-hernandez** [comment]
Does the codecov token need to be a secret or can we hard-code to the github action?

**Comment by rwedge** [comment]
> Does the codecov token need to be a secret or can we hard-code to the github action?

https://docs.codecov.io/docs/about-the-codecov-bash-uploader#upload-token

Based on this documentation, we may not need an upload token

**Comment by jeff-hernandez** [comment]
@rwedge okay will remove the token. I just came across [this github action](https://github.com/codecov/codecov-action) for uploading to codecov.

**Comment by jeff-hernandez** [comment]
I'm getting a [missing token error](https://github.com/alteryx/featuretools/runs/1408990953?check_suite_focus=true#step:9:443) from codecov
> Error: Missing repository upload token

**Comment by jeff-hernandez** [comment]
In the [changelog](https://docs.codecov.io/changelog/release-notes-for-codecov-v4413#new) codecov v4.4.13 mentions tokenless uploading via GitHub Actions:
> [Github] Increases buffer to 60 minutes for tokenless uploading via GitHub Actions in order to support longer running GitHub Actions pipelines.

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -16,9 +16,10 @@ Release Notes
         * Update links to repo in documentation to use alteryx org url (:pr:`1224`)
     * Testing Changes
         * Update release notes check to use new repo url (:pr:`1222`)
+        * Added integration tests as a github action (:pr:`1226`)
```
Comment (rwedge) [inline docs/source/release_notes.rst:19]: Let's change this to "Migrated CI tests to github actions"

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+  win_unit_tests:
+    name: ${{ matrix.python_version }} windows unit tests
+    runs-on: windows-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        python_version: ["3.6", "3.7", "3.8"]
+    steps:
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v2.0.0
```
Comment (rwedge) [inline .github/workflows/tests.yml:215]: should we use v2 instead of v2.0.0

**Comment by rwedge** [comment]
Should this PR also remove the circleci config files?

**Comment by jeff-hernandez** [comment]
> Should this PR also remove the circleci config files?

I think it'd be easier to update the branch protection rules by removing circleci in a separate PR.

File: .github/workflows/tests.yml (hunk 2 above)
Comment (jeff-hernandez) [reply .github/workflows/tests.yml]: updated step to set up miniconda using original code

**Review by rwedge — APPROVED** [review]
Looks good!

**Pull request merged**